client/app/scripts/actions/app-actions.js

```diff
@@ -604,6 +622,31 @@ export function receiveNodesDelta(delta) {
   };
 }
 
+function updateFromNodesDeltaBuffer(dispatch, state) {
+  const isPaused = isPausedSelector(state);
+  const isBufferEmpty = state.get('nodesDeltaBuffer').isEmpty();
+
+  if (isPaused || isBufferEmpty) {
+    dispatch(resetNodesDeltaBuffer());
+  } else {
+    dispatch(receiveNodesDelta(state.get('nodesDeltaBuffer').first()));
+    dispatch({ type: ActionTypes.POP_NODES_DELTA_BUFFER });
+  }
+}
+
+export function clickResumeUpdate() {
+  return (dispatch, getServiceState) => {
+    dispatch({
+      type: ActionTypes.CLICK_RESUME_UPDATE
+    });
+    // Periodically merge buffered nodes deltas until the buffer is emptied.
+    nodesDeltaBufferUpdateTimer = setInterval(
```

Comment:
Yeah, that's a good catch. I did refactor this code here, but I didn't want to change that logic in any way as it doesn't really have to do with this PR. I'd keep it like this until a problem occurs and then we can address it separately in a small PR.